### ISSUE TITLE
Can't install Magit on Emacs 24.3 using package management

### ISSUE THREAD

**Issue by alamaison** [magit/magit#2044]
Since Magit changed to require Emacs 24.3, there is now no way to install Magit on Emacs 24.3 using package management.  Since ELPA doesn't support choosing older versions, everyone is stuck with the same requirement.

Emacs 24.3 is still a very popular version because it is the latest version you can get on Ubuntu 14.04 - the most recent LTS release.

**Comment by tarsius** [comment]
You can also install `1.4.2` by installing it from its Git repository. But since you have to install *something* manually, you might as well let that be Emacs.

Continuing to support `24.3` was not an option. Some things we do now would not have been possible without `24.4` because we needed to customize behavior which in `24.3` was implemented in C.

**Comment by greghendershott** [comment]
Not just emacs. Also git.

I love magit and contribute via GratiPay.

I'm really looking forward to using 2.1.0 .... after I finish building Emacs and Git from source. :)

Seriously, it's a bit bumpy for an Ubuntu user. It might be good to README that?

1. First you need to upgrade to Emacs 24.4+. Fortunately I already had built Emacs from source to get 25.0.50 to test something. So I'd already done that half hour of downloading 500 MB and building. So I just `sudo apt-get remove emacs24 <etc>` and in the Emacs source actually did a full `make install`.

2. Now I can install magit 2.1.0. But it warns, "you have git 1.9.1, requires >= 1.9.4". So now it's 300 MB from `sudo apt-get install libcurl4-gnutls-dev libexpat1-dev gettext libz-dev libssl-dev asciidoc xmlto docbook2x`, a tiny tar download, and another ~20 minutes to `make install git`.

TL;DR: magit is great and 2.1.0 sounds awesome. It might be nice to add a "Using 2.1.0" section on the README and/or in the MELPA description, to warn folks using Ubuntu or whatever, that they have an hour of downloads and source builds ahead of them. That way they can plan to do this when they have enough time.

And again, I'm sorry if this sounds whiny, because, I really think magit is awesome.

**Comment by madscientist** [comment]
FWIW, you can get these via PPA if you don't want to build them yourself.  For 14.04 you can get the latest Emacs here: https://launchpad.net/~ubuntu-elisp/+archive/ubuntu/ppa?field.series_filter=trusty and Git here: https://launchpad.net/~git-core/+archive/ubuntu/ppa?field.series_filter=trusty

Note that "new enough" versions of both Emacs (24.4.1) and Git (2.1.4) are available by default in newer versions of Ubuntu (15.04).

**Comment by greghendershott** [comment]
@madscientist Thanks, that's helpful (as might be obvious, I'm a recent arrival to Ubuntu from OS X).

**Comment by tom-n-pdx** [comment]
The problem is the systems I use are IT maintained Ubuntu boxes on  LTS Ubuntu 14.04. Upgrading to a newer version of Ubuntu isn't a option  - they won't do it system wide. Installing my own emacs isn't an option - don't have permissions to install packages and don't want to maintain my own version.

I understand the desire to use the newer features - just understand that people in a corporate environment maybe a bit behind and limited by what they can use.

For now - I'll have to try using Git to grab the older version.

It's also too bad the package manager doesn't provide a older version sutable for thos of use on a older version.

**Comment by Mellen** [comment]
@madscientist that's the latest emacs for Ubuntu 14.04, which is 24.3.1, not 24.4

**Comment by soupault** [comment]
FWIW, here is how I manage the config:
```
(if (not (eq emacs-major-version 24))
  (use-package magit
    :ensure t
    :defer t))
```
